### PR TITLE
Issue #1829: Fix ImportError from cocotb import simulator

### PR DESCRIPTION
**Pull request by elgorwi**

### CODE REVIEW

File: cocotb/_os_compat.py [hunk 1]
```diff
@@ -14,8 +14,5 @@
 extra_dll_dir = os.path.join(os.path.dirname(__file__), 'libs')
 
 if sys.platform == 'win32' and os.path.isdir(extra_dll_dir):
-    if sys.version_info >= (3, 8):
-        os.add_dll_directory(extra_dll_dir)
-    else:
-        os.environ.setdefault('PATH', '')
-        os.environ['PATH'] += os.pathsep + extra_dll_dir
+    os.environ.setdefault('PATH', '')
```
Comment (eric-wieser) [inline cocotb/_os_compat.py:17]:
```suggestion
    # TODO[gh-1829]: investigate making os.add_dll_directory work here on Python 3.8
    os.environ.setdefault('PATH', '')
```

Comment (elgorwi) [reply cocotb/_os_compat.py:18]: OK, done

File: cocotb/_os_compat.py [hunk 2]
```diff
@@ -14,8 +14,6 @@
 extra_dll_dir = os.path.join(os.path.dirname(__file__), 'libs')
 
 if sys.platform == 'win32' and os.path.isdir(extra_dll_dir):
-    if sys.version_info >= (3, 8):
-        os.add_dll_directory(extra_dll_dir)
-    else:
-        os.environ.setdefault('PATH', '')
-        os.environ['PATH'] += os.pathsep + extra_dll_dir
+	# TODO[gh-1829]: investigate making os.add_dll_directory work here on Python 3.8
```
Comment (ktbarrett) [inline cocotb/_os_compat.py:17]:
```suggestion
    # TODO[gh-1829]: investigate making os.add_dll_directory work here on Python 3.8
```

Comment (elgorwi) [reply cocotb/_os_compat.py]: Spurious tab removed

**Comment by ktbarrett** [comment]
@elgorwi next time could you use [one of these](https://help.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords)?

**Pull request merged**